swap/peer.go

```diff
@@ -103,8 +98,10 @@ func (sp *Peer) handleEmitChequeMsg(ctx context.Context, msg interface{}) error
 
 	// TODO: check serial and balance are higher
 
-	// reset balance to zero, TODO: fix
-	sp.swap.resetBalance(sp.ID())
+	// reset balance by amount
+	// as this is done by the creditor, receiving the cheque, the amount should be negative,
+	// so that updateBalance will calculate balance + amount which result in reducing the peer's balance
+	sp.swap.resetBalance(sp.ID(), 0-int64(cheque.Honey))
```

Comment:
How will we verify that `cheque.Honey` is proportional to the increase in `cheque.Amount`? Is the plan here to convert the honey using the oracle and check that it is roughly the same?

Reply:
There is no clear idea about this and this whole `honey` concept is fresh and might need further iteration.

My suggestion is to use some signed data from the oracle which would basically express "I used this exchange rate" when creating the cheque, and on the other end we just check the signature or something like that. What do you think?

Reply:
How could the oracle sign something? The `ConfigurablePriceOracle` doesn't have any keys which could sign something. If the oracle was an on-chain contract it also wouldn't be able to sign something.

Reply:
Thank you very much for your comments on this one, my blockchain knowledge is limited. We need  to raise these points then if we want to to explore further the oracle idea.

Reply:
I would suggest to do a check on the exchange rate by querying the oracle again here. If your peer used the same oracle (as previously agreed upon) then the exchange rate should be the same as well. If the exchange rate of the cheque is different than the exchange rate which you get back when you query your own oracle, then probably your peer did not use the same oracle as you and that is a reason to disconnect. 
An edge-case is what to do when the rate changes at the time the cheque is traveling from peer to peer.
I don't think this is a clean solution as it will cost extra time and money to do this cheque. However, otherwise I don't see a possibility to verify that the exchange rate given is the same as the exchange rate which should have been given.

Reply:
While thinking about it: Why doesn't the cheque not just only list the amount in currency?

Reply:
I don't have definitive answers yet here. Asking the oracle on the beneficiary side again, seems like a good approach, but then we need to think that in the meanwhile (even if very unlikely) the price might have already changed, making a bullet-proof settlement impossible.

So currently it seems to me an intermediate working solution to have the `Honey` amount in the Cheque, to indicate directly from the Issuer how much in `honey` she adjusted her balance.

The definitive solution will most likely require some change anyway so let's agree on an interim solution.

Reply:
Going forward a better approach might be to send meta information (dependent on the oracle used) on how the price was derived rather than the honey amount. For example if an on-chain oracle was used you could attach the block hash for the block you used to query the oracle, then the other party can decide wether the block was new enough. If the oracle is an external third party than a signed piece of data could be sent instead. 

But for this week I would suggest to leave it as it is and have `Honey` in there as a "hint" (in the message, not in the actual signed cheque), so that nodes are at least sure they are using the same pricing.

Reply:
I think this are great suggestions. I created a new issue for this:

https://github.com/ethersphere/swarm/issues/1598